deepinv/training/trainer.py

```diff
@@ -39,9 +39,17 @@ class Trainer:
     dictionary of the model, ``loss`` the loss history, ``optimizer`` the state dictionary of the optimizer,
     and ``eval_metrics`` the evaluation metrics history.
 
-    - Use `online_measurements=True` when measurements are simulated online from a ground truth returned by the dataloader.
-    - Use `online_measurements=False` when both the ground truth and measurements are returned by the dataloader (and also optionally physics generator params).
-      This could be from a dataset generated using :class:`deepinv.datasets.generate_dataset`.
+    .. tip::
+        If a validation dataloader `eval_dataloader` is provided, the trainer will also **save the best model** according to the
+        first metric in the list, using the following format:
+        ``save_path/yyyy-mm-dd_hh-mm-ss/best_model.pth.tar``.
+
+    Assuming that `x` is the ground-truth reference and `y` is the measurement and `params` is a dict of :ref:`physics parameters <physics_generators>`,
+    the **dataloaders** should return one of the following options:
+
+    1. `(x, y)` or `(x, y, params)`, which requires `online_measurements=False` (default) otherwise `y` will be ignored and new measurements will be generated online.
+    2. `(x)` or `(x, params)`, which requires `online_measurements=True` for generating measurements in an online manner as `y=physics(x)`. Otherwise, first generate a dataset of `(x,y)` with :class:`deepinv.datasets.generate_dataset` and then use option 1 above.
+    3. If you have a dataset of measurements only `(y)` or `(y, params)` you should modify it such that it returns `(torch.nan, y)` or `(torch.nan, y, params)`  and use options 1 or 2 above with `online_measurements=False`.
```

Comment:
How can option 3 use option 2? Maybe leave out the last bit of this sentence and just say 
```suggestion
    3. If you have a dataset of measurements only `(y)` or `(y, params)` you should modify it such that it returns `(torch.nan, y)` or `(torch.nan, y, params)`. Set `online_measurements=False`.
```

Can you also add a warning that if x is ALL nan, i.e. we have option 3, then online_measurements must be false?

Reply:
done